software/database/src/main/java/brooklyn/entity/database/mysql/MySqlClusterImpl.java

```diff
@@ -0,0 +1,445 @@
+                    // We are interested in SERVICE_PROCESS_IS_RUNNING only while haven't come online yet.
+                    // Probably will get several updates while replication is initialized so an additional
+                    // check is needed whether we have already seen this.
+                    Boolean.FALSE.equals(node.getAttribute(MySqlNode.SERVICE_UP)) &&
+                    !Boolean.TRUE.equals(node.getAttribute(NODE_REPLICATION_INITIALIZED))) {
+
+                // Events executed sequentially so no need to synchronize here.
+                if (Boolean.TRUE.equals(node.getAttribute(NODE_REPLICATION_INITIALIZED))) {
+                    return;
+                }
+                ((EntityLocal)node).setAttribute(NODE_REPLICATION_INITIALIZED, Boolean.TRUE);
+
+                DynamicTasks.queueIfPossible(TaskBuilder.builder()
+                        .name("Configure master-slave replication on node")
+                        .body(new InitReplicationTask(cluster, node))
+                        .build())
+                    .orSubmitAsync(node);
+            }
+        }
+
+    }
+
+    // ============= Member Remove =============
+
+    public class MemberRemovedListener implements SensorEventListener<Entity> {
+        @Override
+        public void onEvent(SensorEvent<Entity> event) {
+            MySqlCluster cluster = (MySqlCluster) event.getSource();
+            Entity node = event.getValue();
+            String slaveAddress = cluster.getAttribute(SLAVE_ID_ADDRESS_MAPPING).remove(node.getId());
+            if (slaveAddress != null) {
+                DynamicTasks.queueIfPossible(TaskBuilder.builder()
+                        .name("Remove slave access")
+                        .body(new RemoveSlaveConfigTask(cluster, slaveAddress))
+                        .build())
+                    .orSubmitAsync(cluster);
+            }
+        }
+    }
+
+    public class RemoveSlaveConfigTask implements Runnable {
+        private MySqlCluster cluster;
+        private String slaveAddress;
+
+        public RemoveSlaveConfigTask(MySqlCluster cluster, String slaveAddress) {
+            this.cluster = cluster;
+            this.slaveAddress = validateSqlParam(slaveAddress);
+        }
+
+        @Override
+        public void run() {
+            // Could already be gone if stopping the entire app - let it throw an exception
+            MySqlNode master = (MySqlNode) Iterables.find(cluster.getMembers(), IS_MASTER);
+            String username = validateSqlParam(cluster.getConfig(SLAVE_USERNAME));
+            executeScriptOnNode(master, String.format("DROP USER '%s'@'%s';", username, slaveAddress));
+        }
+
+    }
+
+    // Can't call node.executeScript directly, need to change execution context, so use an effector task
+    private static String executeScriptOnNode(MySqlNode node, String commands) {
+        return node.invoke(MySqlNode.EXECUTE_SCRIPT, ImmutableMap.of("commands", commands)).getUnchecked();
```

Comment:
Perhaps "commands" could live in a `public static String` variable in `MySQLNode` specifying the effector parameter value.